File: src/main/java/com/coffee/shop/api/demo/category/domain/impl/CategoryMapper.java
package com.coffee.shop.api.demo.category.domain.impl;

import com.coffee.shop.api.demo.category.domain.api.dto.CategoryDto;
import com.coffee.shop.api.demo.category.domain.api.model.Category;
import org.mapstruct.Mapper;

@Mapper(componentModel = "spring")
public interface CategoryMapper {

    Category dtoToEntity(CategoryDto dto);

    CategoryDto entityToDto(Category entity);
}

File: src/main/kotlin/com/coffee/shop/api/demo/category/domain/impl/CategoryMapper.kt
package com.coffee.shop.api.demo.category.domain.impl

import com.coffee.shop.api.demo.category.domain.api.dto.CategoryDto
import com.coffee.shop.api.demo.category.domain.api.model.Category
import org.mapstruct.Mapper

@Mapper(componentModel = "spring")
interface CategoryMapper {
    fun dtoToEntity(dto: CategoryDto?): Category?
    fun entityToDto(entity: Category?): CategoryDto?
}